utils/query-state/Get-ConflictingBranches.psm1

```diff
@@ -0,0 +1,57 @@
+Import-Module -Scope Local "$PSScriptRoot/Get-UpstreamBranchMap.psm1"
+
+function Get-LastTouchedBranch {
+	Param(
+		[Parameter(Mandatory)][String] $upstreamMap,
+		[Parameter(Mandatory)][String] $branchName,
+		[Parameter(Mandatory)][String] $file
+	)
+
+	$temp = $upstreamMap[$branchName];
+	$result = git name-rev --name-only @temp (git rev-list -1 "$branchName" -- "$file")
+	return $result.split("~")[0].split("^")[0]
+}
+
+function Get-MergeConflicts([string]$a, [string]$b) {
+	$output = git merge-tree --name-only --write-tree $a $b 
+	$isClean = $LASTEXITCODE -eq 0 # git-merge-tree uses 0 for clean, 1 for not, and <0 for various other errors
+	return [ordered]@{
+		isClean = $isClean;
+		# Just because this list is empty doesn't mean there aren't conflicts
+		conflictList = ($output | Select-Object -Skip 1).Where({ $_ -eq "" }, [System.Management.Automation.WhereOperatorSelectionMode]::Until)
+	}
```

Comment:
To make the conflictList useful, you'll need `--no-messages` to be added. This should also use `Invoke-ProcessLogs`... I'm gonna make one in another PR that you'll be able to leverage here.

Reply:
This uses Where(Predicate, Until) to clear off the messages (though I don't know how git handles not having any messages)

But point taken